docs(Lib/distutils/util.py): strtobool docstring mention case-insensitivity

yxliang01 opened the pull request:
Minor docstring improvement, no bpo or issue needed.

The docstring for function `strtobool` didn't mention that the true values and false values are treated case-insensitively. E.g. `strtobool` treats `True` and `TRUE` are treated same as `true`.

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->

yxliang01 committed:
docs(Lib/distutils/util.py): strtobool docstring mention case-insensitivity

The docstring for function `strtobool` didn't mention that the true values and false values are treated case-insensitively. E.g. `strtobool` treats `True` and `TRUE` are treated same as `true`.

tirkarthi commented on Lib/distutils/util.py:
```diff
@@ -295,8 +295,8 @@ def strtobool (val):
-    are 'n', 'no', 'f', 'false', 'off', and '0'.  Raises ValueError if
-    'val' is anything else.
+    are 'n', 'no', 'f', 'false', 'off', and '0'. Alphabets are treated
+    case-insensitively. Raises ValueError if 'val' is anything else.
```
LGTM. It also has a test so I guess it's worth documenting this. Maybe once accepted the corresponding docs page could also be updated ? 

Docs : https://github.com/python/cpython/blob/master/Doc/distutils/apiref.rst 
Test : https://github.com/python/cpython/blob/79da388a4016e24c4258dcc62cd0fa9dde0acb5b/Lib/distutils/tests/test_util.py#L271

yxliang01 replied:
Good point. Doing it now.

yxliang01 replied:
@tirkarthi See commit https://github.com/python/cpython/pull/12621/commits/f2fe330ed451c33a2c786aad53df6a652b18d61e .

yxliang01 committed:
test(Lib/distutils/tests/test_util.py): improve test_strtobool to explicitly test about the case-insensitivity

Improved the test `test_strtobool` to explicitly test for case-insensitivity that `strtobool` should treat all alphabets case-insensitively.

merwok commented on Lib/distutils/util.py:
```diff
@@ -295,8 +295,8 @@ def strtobool (val):
     True values are 'y', 'yes', 't', 'true', 'on', and '1'; false values
-    are 'n', 'no', 'f', 'false', 'off', and '0'.  Raises ValueError if
-    'val' is anything else.
+    are 'n', 'no', 'f', 'false', 'off', and '0'. Alphabets are treated
```
```suggestion
    are 'n', 'no', 'f', 'false', 'off', and '0'. Letters are treated
```

merwok committed:
docs(Lib/distutils/util.py): improve comment

Co-Authored-By: yxliang01 <yxliang01@users.noreply.github.com>

merwok reviewed (changes requested):
Document in user-facing docs, not internal API reference.

yxliang01 committed:
fix.doc(distutils/apiref.rst): update user-facing documentation for strtobool
